src/librustc_mir/transform/const_prop.rs

```diff
@@ -243,11 +244,14 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for ConstPropMachine {
         l.access()
     }
 
-    fn before_eval_static(
-        _ecx: &InterpCx<'mir, 'tcx, Self>,
-        _place_static: &Static<'tcx>,
+    fn before_access_static(
+        allocation: &Allocation<Self::PointerTag, Self::AllocExtra>,
     ) -> InterpResult<'tcx> {
-        throw_unsup_format!("can't eval statics in ConstProp");
+        if allocation.mutability == Mutability::Mutable {
```

Comment:
for now also bail out if the allocation has relocations. If a static contains a `*mut i32` it may be legal to mutate the memory behind that (if not now, maybe in the future).